inc/saibfd.h

```diff
@@ -0,0 +1,563 @@
+     * @validonly SAI_BFD_SESSION_ATTR_HW_LOOKUP_VALID == false
+     */
+    SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE,
+
+    /**
+     * @brief IP header version
+     *
+     * @type sai_uint8_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_BFD_SESSION_ATTR_IPHDR_VERSION,
+
+    /**
+     * @brief IP header TOS
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_BFD_SESSION_ATTR_TOS,
+
+    /**
+     * @brief IP header TTL
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 255
+     */
+    SAI_BFD_SESSION_ATTR_TTL,
+
+    /**
+     * @brief Source IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_BFD_SESSION_ATTR_SRC_IP_ADDRESS,
+
+    /**
+     * @brief Destination IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_BFD_SESSION_ATTR_DST_IP_ADDRESS,
+
+    /**
+     * @brief Tunnel IP header TOS
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     * @validonly SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE == SAI_BFD_ENCAPSULATION_TYPE_IP_IN_IP
+     */
+    SAI_BFD_SESSION_ATTR_TUNNEL_TOS,
+
+    /**
+     * @brief Tunnel IP header TTL
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 255
+     * @validonly SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE == SAI_BFD_ENCAPSULATION_TYPE_IP_IN_IP
+     */
+    SAI_BFD_SESSION_ATTR_TUNNEL_TTL,
+
+    /**
+     * @brief Tunnel source IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @condition SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE == SAI_BFD_ENCAPSULATION_TYPE_IP_IN_IP
+     */
+    SAI_BFD_SESSION_ATTR_TUNNEL_SRC_IP_ADDRESS,
+
+    /**
+     * @brief Tunnel destination IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @condition SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE == SAI_BFD_ENCAPSULATION_TYPE_IP_IN_IP
+     */
+    SAI_BFD_SESSION_ATTR_TUNNEL_DST_IP_ADDRESS,
+
+    /**
+     * @brief L2 source MAC address
+     *
+     * @type sai_mac_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_BFD_SESSION_ATTR_HW_LOOKUP_VALID == false
+     */
+    SAI_BFD_SESSION_ATTR_SRC_MAC_ADDRESS,
+
+    /**
+     * @brief L2 destination MAC address
+     *
+     * @type sai_mac_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_BFD_SESSION_ATTR_HW_LOOKUP_VALID == false
+     */
+    SAI_BFD_SESSION_ATTR_DST_MAC_ADDRESS,
+
+    /**
+     * @brief To enable echo function on BFD session
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_BFD_SESSION_ATTR_ECHO_ENABLE,
+
+    /**
+     * @brief Multi hop BFD session
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_BFD_SESSION_ATTR_MULTIHOP,
```

Comment:
what is multihop attribute here?

Reply:
This is a type of BFD session where the packet is expected to go over more than one ip hop